hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/DefinitionsService.java

```diff
@@ -38,39 +39,108 @@
+     * manipulated in separate calls. The new <code>Trigger</code> will be persisted.  When fully defined a call to
+     * {@link #updateTrigger(String, Trigger)} is needed to enable the <code>Trigger</code>.
+     * <p>
+     * A non-group trigger can never be made into a group trigger, and vice-versa.
+     * </p>
      * @param tenantId Tenant where trigger is created
      * @param trigger New trigger definition to be added
      * @throws Exception If the <code>Trigger</code> already exists.
+     * @see {@link #addGroupTrigger(String, Trigger)} for adding a group trigger.
      */
     void addTrigger(String tenantId, Trigger trigger) throws Exception;
 
     /**
+     * Create a new Group <code>Trigger</code>. <code>Conditions</code> and <code>Dampening</code> are
+     * manipulated in separate calls. The new <code>Group Trigger</code> will be persisted.  When fully
+     * defined a call to {@link #updateGroupTrigger(String, Trigger)} is needed to enable the <code>Trigger</code>.
+     * <p>
+     * A non-group trigger can never be made into a group trigger, and vice-versa.
+     * </p>
+     * @param tenantId Tenant where trigger is created
+     * @param groupTrigger New trigger definition to be added
+     * @throws Exception If the <code>Trigger</code> already exists.
+     * @see {@link #addTrigger(String, Trigger)} for adding a non-group trigger.
+     */
+    void addGroupTrigger(String tenantId, Trigger groupTrigger) throws Exception;
+
+    /**
+     * Generate a member trigger for the specified group trigger. The dataIdMap replaces the tokens in the
+     * group trigger's conditions with actual dataIds. The member trigger gets the enabled state of the group.
+     * @param tenantId Tenant where trigger is stored
+     * @param groupId Group triggerId from which to spawn the member trigger
+     * @param memberId The member triggerId, unique id within the tenant, if null an Id will be generated
+     * @param memberName The member triggerName, not null, unique name within the tenant
+     * @param memberContext The member triggerContext. If null the context is inherited from the group trigger
+     * @param dataIdMap Tokens to be replaced in the new trigger
+     * @return the member trigger
+     * @throws Exception on any problem
+     * @see {@link #addTrigger(String, Trigger)} for adding a non-group trigger.
+     * @see {@link #addGroupTrigger(String, Trigger)} for adding a group trigger.
+     */
+    Trigger addMemberTrigger(String tenantId, String groupId, String memberId, String memberName,
+            Map<String, String> memberContext, Map<String, String> dataIdMap) throws Exception;
+
+    /**
      * The <code>Trigger</code> will be removed from the Alerts engine, as needed, and will no longer be persisted.
      * @param tenantId Tenant where trigger is stored
      * @param triggerId Trigger to be removed
+     * @throws NotFoundException if trigger is not found
```

Comment:
I am not opposed to adding an analogous FoundException. These need to be @ApplicationException because we don't want then to be wrapped as EJBException when thrown from the EJB method, and then can be caught easily by the caller (in this case the Rest API code).  Because we now must fetch the Trigger in most, if not all, EJB methods (to perform proper checking for group/member/orphan/standard settings) , it was more efficient and cleaner to introduce the new Exception.

I will see if I can make your recommended changes.  If I have difficulty this can be post PR work.